src/node_api.cc

```diff
@@ -706,6 +706,40 @@ bool FindWrapper(v8::Local<v8::Object> obj,
   return true;
 }
 
+static void DeleteEnv(const v8::WeakCallbackInfo<napi_env__>& data) {
+  delete data.GetParameter();
+}
+
+napi_env get_env(v8::Local<v8::Context> context) {
+  napi_env result;
+
+  auto isolate = context->GetIsolate();
+  auto global = context->Global();
+
+  auto maybe_key = v8::String::NewFromOneByte(isolate,
+      reinterpret_cast<const uint8_t *>("N-API Environment"),
+      v8::NewStringType::kInternalized);
+  CHECK(!maybe_key.IsEmpty());
+  auto key = v8::Private::ForApi(isolate, maybe_key.ToLocalChecked());
+
+  auto maybe_value = global->GetPrivate(context, key);
+  CHECK(!maybe_value.IsEmpty());
+  auto value = maybe_value.ToLocalChecked();
```

Comment:
Just an observation but .ToLocalChecked() also does the equivalent of `CHECK(!maybe_value.IsEmpty())`.